lib/api/core/notifier.js

```diff
@@ -258,8 +258,42 @@ class NotifierController {
    * @param {Array} ids - list of deleted document IDs
    */
   notifyDocumentDelete(request, ids) {
-    return Bluebird.each(ids, (id, callback) => {
-      return this.kuzzle.services.list.internalCache.search(this.cacheKeyPrefix + id)
+    return Bluebird.all(ids.map(id => {
+      const
+        getRequest = new Request({
```

Comment:
I don't see why the content of a _deleted_ document should be sent: what's the point, since it has been deleted?
The only users interested by receiving deleted documents notifications should already have their ID stored somewhere in order to sync their app and remove the deleted document.
So to me, we should only send the document id.

Other than that, I agree with the way @benoitvidis describes how the deletion notification function should behave.